quantum/rgb_matrix_animations/breathing_anim.h

```diff
@@ -0,0 +1,19 @@
+#pragma once
+#ifndef DISABLE_RGB_MATRIX_BREATHING
+
+extern rgb_config_t rgb_matrix_config;
+
+bool rgb_matrix_breathing(bool init, uint8_t iter) {
+  RGB_MATRIX_USE_LIMITS(led_min, led_max);
+
+  uint16_t time = scale16by8(g_rgb_counters.tick, rgb_matrix_config.speed / 8);
+  uint8_t val = scale8(abs8(sin8(time) - 128) * 2, rgb_matrix_config.val);
+  HSV hsv = { rgb_matrix_config.hue, rgb_matrix_config.sat, val };
+  RGB rgb = hsv_to_rgb(hsv);
+  for (uint8_t i = led_min; i < led_max; i++) {
```

Comment:
For some reason, each iteration of this loop is taking about 0.6ms, so total time taken is around `DRIVER_LED_TOTAL * 0.6ms`. I can see it as a choppy animation on the keyboard.

Reply:
`DRIVER_LED_TOTAL * 0.6ms` or `RGB_MATRIX_LED_PROCESS_LIMIT * 0.6ms`?
Either way that is super odd. I'll do another pass on this animation tonight as I'd like to get it to pause at the high end vs continual fluctuations as it does now.

Reply:
DRIVER_LED_TOTAL, as in 119 * 0.6ms = 71.4ms for a breathing animation on the CTRL

Reply:
ew, will fix it tonight. was coding blind last night without testing the results, my bad

Reply:
I think the goal would be to get this under 33ms and aim for a steady 30fps. The closer you can get to 16.6ms/60fps the better it should look.

Reply:
This animation was running at 0.6ms for the entire update (119 LEDs processed) in my bench marking from a few commits back before it was updated to batch processing. But now I see that simply the call to rgb_matrix_set_color_all() is taking that much time and then happening 119 times. I briefly looked into the logic myself and the compiled assembly but couldn't see anything fishy with how the loop was being assembled. I didn't try to compile without optimizations though, which might be a useful test.

Reply:
...I'm dumb, that should have been changed to rgb_matrix_set_color(i... instead of rgb_matrix_set_color_all

Reply:
there, fixed. Now I'm not doing 119 * (119 / 15) iterations....